src/components/specialQuestions/Textile.tsx

```diff
@@ -0,0 +1,92 @@
+'use client'
+
+import Question from '@/components/form/Question'
+import Trans from '@/components/translation/trans/TransClient'
+import Button from '@/design-system/buttons/Button'
+import { useRule } from '@/publicodes-state'
+import type { DottedName } from '@incubateur-ademe/nosgestesclimat'
+import { utils } from 'publicodes'
+import { useEffect } from 'react'
+import { twMerge } from 'tailwind-merge'
+import PencilIcon from '../icons/PencilIcon'
+
+type Props = { question: DottedName }
+export default function Textile({ question, ...props }: Props) {
+  const { numericValue: totalPiecesTextile } = useRule(
+    'divers . textile . nombre total'
+  )
+  const { value: preciseChoice, setValue: setPreciseChoice } = useRule(
+    'divers . textile . choix précis'
+  )
+  const { setValue } = useRule(question)
+
+  const possibleAnswers: Record<string, DottedName> = {
+    minimum: 'divers . textile . volume . minimum',
+    'renouvellement occasionnel':
+      'divers . textile . volume . renouvellement occasionnel',
+    'accro au shopping': 'divers . textile . volume . accro au shopping',
+  }
+
+  useEffect(() => {
+    if (preciseChoice && totalPiecesTextile > 0) {
```

Comment:
Pareil ici, il vaut mieux faire un early return que d'imbriquer des if, globalement imbriquer des ifs c'est no good bro

```suggestion
    if (!preciseChoice || totalPiecesTextile < 0) return
    
          if (totalPiecesTextile <= 15) {
        setValue(utils.nameLeaf(possibleAnswers['minimum']))
      }
      if (totalPiecesTextile > 15 && totalPiecesTextile <= 35) {
        setValue(utils.nameLeaf(possibleAnswers['renouvellement occasionnel']))
      }
      if (totalPiecesTextile > 35) {
        setValue(utils.nameLeaf(possibleAnswers['accro au shopping']))
      }
    
```